assessment_software/src/app/textarea-component/textarea.component.ts

```diff
@@ -0,0 +1,18 @@
+import { Component, Input, Output, EventEmitter } from "@angular/core";
+
+@Component({selector:'app-textarea',
+templateUrl:'./textarea.component.html',
+styleUrls:['./textarea.component.css'],
+})
+
+export class TextareaComponent{
+    @Input () styleClass:string='textarea-skin';
+    @Input() maxlength:number=0;
+    @Input() minlength:number=0;
+    @Output() value= new EventEmitter<string>();
```

Comment:
It is better to rename the variable name "value" to "onValueChange", since it is a callback. Because when you use it on AppCompnent, you would attach a callback function like this:
```

<app-textarea
[maxlength]=30
[minlength]=10
(value)="textAreaOutput($event)" // <---- attaching function(textAreaOutput) to be called on value change
></app-textarea>
```